src/hardware/hardware.cpp

```diff
@@ -367,22 +367,18 @@ void CAPTURE_AddImage(Bitu width, Bitu height, Bitu bpp, Bitu pitch, Bitu flags,
 				PNG_COMPRESSION_TYPE_DEFAULT, PNG_FILTER_TYPE_DEFAULT);
 		}
 #ifdef PNG_TEXT_SUPPORTED
-		int fields = 1;
-		png_text text[1] = {};
-		const char* text_s = "dosbox-staging " VERSION;
-		const size_t text_len = strlen(text_s);
-		char* ptext_s = new char[text_len + 1];
-		strncpy(ptext_s, text_s, text_len);
-		char software[9] = { 'S','o','f','t','w','a','r','e',0};
-		text[0].compression = PNG_TEXT_COMPRESSION_NONE;
-		text[0].key  = software;
-		text[0].text = ptext_s;
-		png_set_text(png_ptr, info_ptr, text, fields);
+		constexpr char keyword[] = "Software";
+		constexpr char value[] = "dosbox-staging " VERSION;
+		constexpr int num_text = 1;
+		static_assert(sizeof(keyword) < 80, "libpng limit");
+		png_text texts[num_text] = {};
+		texts[0].compression = PNG_TEXT_COMPRESSION_NONE;
+		texts[0].key = const_cast<png_charp>(keyword);
+		texts[0].text = const_cast<png_charp>(value);
+		texts[0].text_length = sizeof(value);
```

Comment:
Could populate `texts` at compile-time by making it `constexpr` too (using named-designators so it's still as readable):
``` c++
constexpr png_text texts[] = { {
    .compression = PNG_TEXT_COMPRESSION_NONE,
    .key = const_cast<png_charp>(keyword),
    .text = const_cast<png_charp>(value),
    .text_length = sizeof(value) } };
```

Reply:
Not sure if named designators are going to work in MSVC (they are C thing and work in gnuc++). IIRC they were supposed to add it to C++, but I don't remember when.

I'll try it :)